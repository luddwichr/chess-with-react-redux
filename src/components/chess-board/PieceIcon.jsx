import React from 'react';
import PropTypes from 'prop-types';
import ChessPieces from './chess-pieces.svg';

const PieceIcon = ({player, piece}) => (
  <svg style={{display: 'block'}} viewBox="0 0 44 44">
    <use xlinkHref={`${ChessPieces}#icon-${player}-${piece}`} />
  </svg>
);

PieceIcon.propTypes = {
  player: PropTypes.oneOf(['black', 'white']).isRequired,
  piece: PropTypes.oneOf(['king', 'queen', 'rook', 'bishop', 'knight', 'pawn']).isRequired,
};

export default PieceIcon;
