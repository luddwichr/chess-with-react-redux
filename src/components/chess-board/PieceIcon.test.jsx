import {create} from 'react-test-renderer';
import React from 'react';
import PieceIcon from './PieceIcon';

describe('PieceIcon', () => {
  [
    {player: 'white', piece: 'king'},
    {player: 'white', piece: 'queen'},
    {player: 'white', piece: 'rook'},
    {player: 'white', piece: 'bishop'},
    {player: 'white', piece: 'knight'},
    {player: 'white', piece: 'pawn'},
    {player: 'black', piece: 'king'},
    {player: 'black', piece: 'queen'},
    {player: 'black', piece: 'rook'},
    {player: 'black', piece: 'bishop'},
    {player: 'black', piece: 'knight'},
    {player: 'black', piece: 'pawn'},
  ].forEach(({player, piece}) => {
    it(`should display a ${player} ${piece}`, () => {
      const element = create(<PieceIcon player={player} piece={piece} />);
      expect(element).toMatchSnapshot();
    });
  });
});
