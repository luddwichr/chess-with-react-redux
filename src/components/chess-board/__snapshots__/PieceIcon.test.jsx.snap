// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PieceIcon should display a black bishop 1`] = `
<svg
  style={
    Object {
      "display": "block",
    }
  }
  viewBox="0 0 44 44"
>
  <use
    xlinkHref="undefined#icon-black-bishop"
  />
</svg>
`;

exports[`PieceIcon should display a black king 1`] = `
<svg
  style={
    Object {
      "display": "block",
    }
  }
  viewBox="0 0 44 44"
>
  <use
    xlinkHref="undefined#icon-black-king"
  />
</svg>
`;

exports[`PieceIcon should display a black knight 1`] = `
<svg
  style={
    Object {
      "display": "block",
    }
  }
  viewBox="0 0 44 44"
>
  <use
    xlinkHref="undefined#icon-black-knight"
  />
</svg>
`;

exports[`PieceIcon should display a black pawn 1`] = `
<svg
  style={
    Object {
      "display": "block",
    }
  }
  viewBox="0 0 44 44"
>
  <use
    xlinkHref="undefined#icon-black-pawn"
  />
</svg>
`;

exports[`PieceIcon should display a black queen 1`] = `
<svg
  style={
    Object {
      "display": "block",
    }
  }
  viewBox="0 0 44 44"
>
  <use
    xlinkHref="undefined#icon-black-queen"
  />
</svg>
`;

exports[`PieceIcon should display a black rook 1`] = `
<svg
  style={
    Object {
      "display": "block",
    }
  }
  viewBox="0 0 44 44"
>
  <use
    xlinkHref="undefined#icon-black-rook"
  />
</svg>
`;

exports[`PieceIcon should display a white bishop 1`] = `
<svg
  style={
    Object {
      "display": "block",
    }
  }
  viewBox="0 0 44 44"
>
  <use
    xlinkHref="undefined#icon-white-bishop"
  />
</svg>
`;

exports[`PieceIcon should display a white king 1`] = `
<svg
  style={
    Object {
      "display": "block",
    }
  }
  viewBox="0 0 44 44"
>
  <use
    xlinkHref="undefined#icon-white-king"
  />
</svg>
`;

exports[`PieceIcon should display a white knight 1`] = `
<svg
  style={
    Object {
      "display": "block",
    }
  }
  viewBox="0 0 44 44"
>
  <use
    xlinkHref="undefined#icon-white-knight"
  />
</svg>
`;

exports[`PieceIcon should display a white pawn 1`] = `
<svg
  style={
    Object {
      "display": "block",
    }
  }
  viewBox="0 0 44 44"
>
  <use
    xlinkHref="undefined#icon-white-pawn"
  />
</svg>
`;

exports[`PieceIcon should display a white queen 1`] = `
<svg
  style={
    Object {
      "display": "block",
    }
  }
  viewBox="0 0 44 44"
>
  <use
    xlinkHref="undefined#icon-white-queen"
  />
</svg>
`;

exports[`PieceIcon should display a white rook 1`] = `
<svg
  style={
    Object {
      "display": "block",
    }
  }
  viewBox="0 0 44 44"
>
  <use
    xlinkHref="undefined#icon-white-rook"
  />
</svg>
`;
